library/src/blas2/rocblas_tbmv.hpp

```diff
@@ -258,52 +258,46 @@ __global__ void tbmvx_kernel(rocblas_operation transA,
 /**
   *  First, makes a copy of 'x', then uses a modified gemv algorithm
   *  to perform x := transA(A) * x_copy
-  *  x_copy should be of size sizeof(T) * m bytes.
+  *  x_copy should be of size sizeof(T) * m bytes * batch_count.
+  *
+  *  Here, U is either a `const T* const*` or a `const T*`
+  *  V is either a `T*` or a `T* const*`
   */
-template <typename T>
+template <typename T, typename U, typename V>
 rocblas_status rocblas_tbmv_template(rocblas_handle    handle,
                                      rocblas_fill      uplo,
                                      rocblas_operation transA,
                                      rocblas_diagonal  diag,
                                      rocblas_int       m,
                                      rocblas_int       k,
-                                     const T*          A,
+                                     U                 A,
                                      rocblas_int       offseta,
                                      rocblas_int       lda,
                                      rocblas_stride    strideA,
-                                     T*                x,
+                                     V                 x,
                                      rocblas_int       offsetx,
                                      rocblas_int       incx,
                                      rocblas_stride    stridex,
                                      rocblas_int       batch_count,
-                                     T*                x_copy)
+                                     V                 x_copy)
 {
     // quick return
     if(!m || !batch_count)
         return rocblas_status_success;
 
-    // in case of negative inc shift pointer to end of data for negative indexing tid*inc
-    offsetx = incx < 0 ? offsetx - ptrdiff_t(incx) * (m - 1) : offsetx;
-
     // First we make a copy of x so we can avoid RAW race conditions in the kernel
     int  copy_blocks = (m - 1) / 256 + 1;
     dim3 copy_grid(copy_blocks, batch_count);
     dim3 copy_threads(256);
 
-    hipLaunchKernelGGL(copy_kernel,
-                       copy_grid,
-                       copy_threads,
-                       0,
-                       handle->rocblas_stream,
-                       m,
-                       x,
-                       offsetx,
-                       incx,
-                       stridex,
-                       x_copy,
-                       0,
-                       1,
-                       m);
+    rocblas_status status = rocblas_copy_template<256>(
+        handle, m, x, offsetx, incx, stridex, x_copy, 0, 1, m, batch_count);
+
+    if(status != rocblas_status_success)
+        return status;
+
+    // in case of negative inc shift pointer to end of data for negative indexing tid*inc
+    offsetx = incx < 0 ? offsetx - ptrdiff_t(incx) * (m - 1) : offsetx;
```

Comment:
Sorry if this is a duplicate, but my comment got lost:  offsetx is a rocblas_int which may too small for the result of ptr_diff_t math.

Reply:
Good catch, fixed.